elements/lisk-p2p/src/peer/inbound.ts

```diff
@@ -46,13 +51,25 @@ export class InboundPeer extends Peer {
 			this.emit(EVENT_INBOUND_SOCKET_ERROR, error);
 		};
 		this._handleInboundSocketClose = (code, reason) => {
+			if (this._pingIntervalId) {
+				clearInterval(this._pingIntervalId);
+			}
 			this.emit(EVENT_CLOSE_INBOUND, {
 				peerInfo,
 				code,
 				reason,
 			});
 		};
+		this._pingStart = Date.now();
+		this._handlePong = () => {
+			const latency = this._pingStart - Date.now();
+			this._latency = latency;
```

Comment:
A malicious peer could ignore the first ping, and then emit a pong at the exact same time as it expects us to send the next ping (so basically it emits the pong before it actually received the ping) so it could get a low latency score. Alternatively, the peer could just constantly spam us with pongs every few milliseconds.

We could:
1. Randomize the ping interval a bit.
2. Add a callback to the `socket.emit(event, callback)` and then the peer just sends a response instead of emitting a separate pong; that way each response (pong) is coupled to its request (ping) so it cannot be gamed by spamming pongs.

Reply:
It's a lot more difficult to game than before though.